core/src/OpenACC/Kokkos_OpenACC.cpp

```diff
@@ -46,25 +46,47 @@
 
 #include <OpenACC/Kokkos_OpenACC.hpp>
 #include <OpenACC/Kokkos_OpenACC_Instance.hpp>
+#include <OpenACC/Kokkos_OpenACC_Traits.hpp>
 #include <impl/Kokkos_Profiling.hpp>
 #include <impl/Kokkos_ExecSpaceManager.hpp>
+#include <impl/Kokkos_DeviceManagement.hpp>
 
 #include <ostream>
 
 Kokkos::Experimental::OpenACC::OpenACC()
-    : m_space_instance(Impl::OpenACCInternal::singleton()) {}
+    : m_space_instance(
+          &Kokkos::Experimental::Impl::OpenACCInternal::singleton(),
+          [](Impl::OpenACCInternal*) {}) {
+  Impl::OpenACCInternal::singleton().verify_is_initialized(
+      "OpenACC instance constructor");
+}
+
+Kokkos::Experimental::OpenACC::OpenACC(int async_arg)
+    : m_space_instance(new Kokkos::Experimental::Impl::OpenACCInternal,
+                       [](Impl::OpenACCInternal* ptr) {
+                         ptr->finalize();
+                         delete ptr;
+                       }) {
+  Impl::OpenACCInternal::singleton().verify_is_initialized(
+      "OpenACC instance constructor");
+  m_space_instance->initialize(Impl::OpenACCInternal::singleton().m_accDev,
+                               async_arg);
+}
 
 void Kokkos::Experimental::OpenACC::impl_initialize(
     InitializationSettings const& settings) {
-  Impl::OpenACCInternal::singleton()->initialize(settings);
+  acc_init(Kokkos::Experimental::Impl::OpenACC_Traits::dev_type);
+  Impl::OpenACCInternal::singleton().initialize(
+      Kokkos::Impl::get_gpu(settings));
 }
 
 void Kokkos::Experimental::OpenACC::impl_finalize() {
-  Impl::OpenACCInternal::singleton()->finalize();
+  Impl::OpenACCInternal::singleton().finalize();
+  acc_shutdown(Kokkos::Experimental::Impl::OpenACC_Traits::dev_type);
```

Comment:
`acc_shutdown_device` is also another latest feature introduced in the OpenACC Version 3.2, and thus there exists no OpenACC compiler implementation supporting this feature yet.

Reply:
OK about the preferring `acc_shutdown` over `acc_shutdown_device` part.
What about interoperability?  Is not going to be an issue that Kokkos calls `acc_shutdown`?

Reply:
Inserting explicit `acc_init` and acc_shutdown` APIs mean that the Kokkos program will serve as the main program in terms of the interoperability with other OpenACC program. (The Kokkos program will explicitly initialize and shutdown the OpenACC runtime, and other OpenACC code can be inserted only during the execution of the Kokkos program.)
In fact, explicit `acc_init` and `acc_shutdown` API calls are not necessary; if we remove these calls from the Kokkos backend, it will mean that other OpenACC program can serve as the main program controlling when to initialize and/or shutdown the OpenACC runtime.

Reply:
Deleted `acc_init` and `acc_shutdown` API calls.